### PR TITLE
Form composition pattern

### PR DESCRIPTION
**Pull request by xredo**
This code shows the form composition pattern the way @oriolbcn implemented it in one of our internal projects.

### CODE REVIEW

File: rails/form-composition-pattern/app/controllers/profiles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      render :edit
+    end
+  end
+
+  private
+
+  def profile_params
+    params.require(:profile_form).permit(
+      user_attributes: %i(email),
+      organization_attributes: [:name]
```
Comment (oriolbcn) [inline rails/form-composition-pattern/app/controllers/profiles_controller.rb]: `%i(name)`

File: rails/form-composition-pattern/app/controllers/signups_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      render :new
+    end
+  end
+
+  private
+
+  def signup_params
+    params.require(:sign_up_form).permit(
+      user_attributes: %i(email password),
+      organization_attributes: [:name]
```
Comment (oriolbcn) [inline rails/form-composition-pattern/app/controllers/signups_controller.rb]: `%i(name)`

File: rails/form-composition-pattern/app/forms/signup_form.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# This pattern is used to have multiple models being saved in the same form without
+# writing additional logic that the one that the one Rails already provides to us.
```
Comment (oriolbcn) [inline rails/form-composition-pattern/app/forms/signup_form.rb]: `that the one` is repeated twice: `that the one that the one`

**Review by oriolbcn — CHANGES_REQUESTED** [review]
a typo and a style consistency change ;)

**Comment by xredo** [comment]
@oriolbcn Thanks for the review. Changes applied.

**Review by oriolbcn — APPROVED** [review]

**Review by artero — APPROVED** [review]

**Review by ItsGenis — APPROVED** [review]

**Pull request merged**